content/open-source/concepts/graphql/graphql.md

```diff
@@ -0,0 +1,38 @@
+---
+Title: 'GraphQL'
+Description: 'GraphQL is an open-source query language that allows developers to consume, construct, and create APIs in their own unique way.'
+Subjects:
+  - 'Computer Science'
+  - 'Web Development'
+Tags:
+  - 'APIs'
+  - 'Open Source'
+  - 'Queries'
+  - 'Servers'
+CatalogContent:
+  - 'paths/computer-science'
+  - 'paths/back-end-engineer-career-path'
+  - 'paths/front-end-engineer-career-path'
+---
+
+[**GraphQL**](https://graphql.org/) is an open-source query language that allows developers to consume, construct, and create [APIs](https://www.codecademy.com/resources/docs/general/api) in their own unique way.
+
+## History of GraphQL
+
+GraphQL was initially developed by Meta (formerly Facebook) in 2012 to address internal issues with their mobile applications. As an alternative to representational state transfer (REST) architecture, GraphQL became open source in 2015. Consequently, because GraphQL makes data transfers more efficient, it reduced network usage for Meta's mobile applications drastically. Since then, GraphQL has attracted a large number of developers and is becoming increasingly popular for client-server communication.
+
+## Why Adopt GraphQL?
+
+Before discussing GraphQL, developers should be familiar with RESTful architecture. One of REST's disadvantages is it can result in an excessive number of [HTTP](https://www.codecademy.com/resources/docs/general/http) queries or too much data being fetched. Alternatively, GraphQL allows developers to ask for specific data instead of fetching a large object with all of the data every time. There is no room for errors like underfetching and overfetching with GraphQL. Finally, GraphQL allows developers to easily integrate with [databases](https://www.codecademy.com/resources/docs/general/database), [JSON](https://www.codecademy.com/resources/docs/general/json) files, RESTful APIs, and cloud services.
```

Comment:
- First sentence, should we use "adopting" rather than "discussing"
- Would it also be helpful to put these in bullet points to make each point clearly laid out?